spark-bigquery-dsv2/spark-3.1-bigquery/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryTable.java

```diff
@@ -44,24 +48,55 @@ public class BigQueryTable implements Table, SupportsRead, SupportsWrite {
           TableCapability.BATCH_READ, TableCapability.BATCH_WRITE, TableCapability.TRUNCATE);
 
   private Injector injector;
-  private SparkBigQueryConfig config;
-  private StructType schema;
   private TableInfo tableInfo;
+  private StructType schema;
 
-  public BigQueryTable(Injector injector, StructType sparkProvidedSchema) {
+  private BigQueryTable(Injector injector, TableInfo tableInfo, StructType schema) {
     this.injector = injector;
-    this.config = injector.getInstance(SparkBigQueryConfig.class);
+    this.tableInfo = tableInfo;
+    this.schema = schema;
+  }
+
+  public static BigQueryTable fromConfigurationAnsSchema(
```

Comment:
Minor typo: Should be `fromConfigurationAndSchema` rather than `fromConfigurationAnsSchema`